doc/src/build/custom-upgrade-policy.md

```diff
@@ -243,19 +195,91 @@ module policy::day_of_week {
 }
 ```
 
-We then need to add a function to authorize an upgrade, if we are on
-the correct day of the week:
+This code includes a constructor and defines the object type for the custom upgrade policy.
+
+You then need to add a function to authorize an upgrade, if on the correct day of the week. First, define a couple of constants, one for a day of the week where the upgrade isn't allowed, and another to define the number of milliseconds in a day (to be used shortly). Add these definitions directly under the current `ENotWeekDay` one.
+
+```rust
+// Request to authorize upgrade on the wrong day of the week.
+const ENotAllowedDay: u64 = 2;
+
+const MS_IN_DAY: u64 = 24 * 60 * 60 * 1000;
+```
+
+After the `new_policy` function, add an `authorize_upgrade` function to check that the day of the week is not violating the policy.
+
+```rust
+public fun authorize_upgrade(
+    cap: &mut UpgradeCap,
+    policy: u8,
+    digest: vector<u8>,
+    ctx: &TxContext,
+): package::UpgradeTicket {
+    assert!(week_day(ctx) == cap.day, ENotAllowedDay);
+    package::authorize_upgrade(&mut cap.cap, policy, digest)
+}
+```
+
+The signature of a custom `authorize_upgrade` can be different from the signature of `sui::package::authorize_upgrade` as long as it returns an `UpgradeTicket`.
+
+Next, add a `week_day` function to get the current weekday. As promised, the function uses the `MS_IN_DAY` constant you defined earlier.
```

Comment:
Note that `authorize_upgrade` calls `week_day`, so if we are splitting them into their own code blocks, we may want to put this one first?